### PR TITLE
[Accommodation]: added AccommodationController with the corresponding logic

### PR DESCRIPTION
**Pull request by annton-shakur**
## Added:
* AccommodationController
* AccommodationService with the implementation
* AccommodationRepository with custom queries

## Changed:
* AccommodationResponseDto (replaced List<String> amenitiesNames with List<Long> amenitiesIds)
* UpdateAccommodationRequestDto (replaced List<String> amenitiesNames with List<Long> amenitiesIds)
* CreateAccommodationRequestDto (replaced List<String> amenitiesNames with List<Long> amenitiesIds)
* AccommodationMapper to set amenityIds instead of amenityNames
* Amenity model (added constructor with id)
* Accommodation model (fixed the typo in the @SQLDelete body)

### CODE REVIEW

**Review by TarasYurkovskyi — APPROVED** [review]
GJ!

File: src/main/java/com/example/accommodationbookingapp/controller/AccommodationController.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+@Tag(name = "Accommodations manager", description = "Endpoint for managing accommodations")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/accommodations")
+public class AccommodationController {
+    private final AccommodationService accommodationService;
+
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping
+    public AccommodationResponseDto save(
```
Comment (vadymhrnk) [inline src/main/java/com/example/accommodationbookingapp/controller/AccommodationController.java:33]: I think you need to put `@PreAuthorize("hasRole('ADMIN')")` here

**Review by vadymhrnk — APPROVED** [review]
Good job. My only suggestion is to add `@PreAuthorize` to endpoints

**Review by MinoDentori — APPROVED** [review]
Looks good to me

**Pull request merged**